### PR TITLE
[Flexible Layouts] Flexible Layout styling fixes

### LINKED ISSUE #6693: Element styling unusable in Stacked Plots and Flex Layouts
#### Summary
First surfaced during VIPER sims as the NIRVSS position tried to setup displays for their position. In short, 
1. Styling in Flex Layouts and Stacked Plots is seriously broken. 
2. The user cannot easily select the parent Stacked Plot after having selected one of its children. 

Putting this altogether in a single issue as the two views seem to share some common problems. 

### Flexible Layouts
1. Add several telemetry points to the layout, then select one and attempt to give it a style. The Style panel lets you think you can, but no style is applied:
Screen Shot 2023-05-24 at 4 13 43 PM

3. Select a container, in this example a column. Apply a style to that, and observe that only the _first_ element in that container is styled: 
Screen Shot 2023-05-24 at 4 12 56 PM

4. Clicking another container in the layout will display the colors associated with the settings from step 2 above, even though another container has been selected. Changing the style here will affect the _first_ element in the _first_ container.
Screen Shot 2023-05-24 at 4 13 57 PM

5. Saving, navving away and back will reveal that the entire Flex Layout has been styled: 
Screen Shot 2023-05-24 at 4 18 34 PM

6. Coming back into Edit, and trying to style the whole Flex Layout ends up apparently styling the _first_ element, as in step 2 above; but wait...
Screen Shot 2023-05-24 at 4 43 56 PM

7. Saving, navving away and back will reveal (you guessed it) that the entire Flex Layout has been styled again, like 4.
8. WORST OF ALL, editing the layout and attempting to set the styles to "No Style" (particularly the text color) results in all text elements **being set to a color that's completely unreadable**:
Screen Shot 2023-05-24 at 5 15 16 PM
Screen Shot 2023-05-24 at 5 20 40 PM

It is now _impossible_ to reset the styling; the user can't even hard-code a different color onto the Flex Layout elements. At this point, the whole view is trashed.

### Stacked Plots
1. Add several telemetry points to the plot, but don't select one of them in the plot area. Attempt to apply a style to the whole domain object; observe the same problem as in Flex Layout step 5 above.
Screen Shot 2023-05-24 at 4 27 48 PM

2. Select the child plot that appears to be styled. Note that the Styles tab indicates no style is applied:
Screen Shot 2023-05-24 at 4 28 38 PM

3. Select an unstyled child plot in the plot area and attempt to give it a style. The Style panel lets you think you can, but no style is applied; this is the same problem as seen in Flex Layouts step 1:
Screen Shot 2023-05-24 at 4 28 54 PM

4. Saving, navving away and back will reveal that (unlike the Flex Layout) _no_ styles have been applied anywhere: 
Screen Shot 2023-05-24 at 4 33 52 PM

5. Coming back into Edit reveals that the the Styles tab has retained the style that was "applied" even though no styling is visible:
Screen Shot 2023-05-24 at 5 05 20 PM

6. When editing and having selected a child plot element, the user cannot click anywhere to select the parent again. The only way to do it is to carefully click on a plot legend name. 

#### Expected vs Current Behavior

1. The Styles tab MUST recognize if an element can be styled or not. If it can't, it must have a disabled or other state that includes a message like "This element cannot be styled".
2. The overall Flex Layout or Stacked Plot itself MUST be style-able.
3. Individual child elements of Flex Layouts and Stacked Plots (telem points, child Overlay Plots, etc.) MUST BE STYLE-ABLE. If an overall style has been applied to the parent layout or plot, the individual styling should be able to coexist with that.
4. Flex Layout containers should NOT be style-able.
5. When the user chooses the "no style" option, REMOVE THE STYLE TAG - do not add a style attribute with a hard-coded color.
6. Conditional Styling MUST work everywhere that static styling does in the same way it does elsewhere.
7. Users MUST be able to click on the empty area above the plot and legend areas to select the parent Stacked Plot container.

#### Steps to Reproduce

See above.

#### Environment

* Open MCT Version:  2.2.0 
* Deployment Type:  VIPER MMOC
* OS: all
* Browser: all

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [X] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [X] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by khalidadil**
Closes #6693

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: e2e/tests/visual-a11y/styling.visual.spec.js [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/**
+ * This test is dedicated to test styling of various plugins
+ */
+
```
Comment (khalidadil) [inline e2e/tests/visual-a11y/styling.visual.spec.js:26]: I created this file as a general one that could be used for all "styling" related tests. Should I instead make this plugin specific (flexible layouts in this case)?

File: src/plugins/flexibleLayout/components/FlexibleLayout.vue [hunk 2]
```diff
@@ -29,7 +29,7 @@
     </div>
 
     <div
-      class="c-fl__container-holder"
+      class="c-fl__container-holder u-style-receiver js-style-receiver"
```
Comment (khalidadil) [inline src/plugins/flexibleLayout/components/FlexibleLayout.vue:32]: Added to fix styles not being applied properly

File: src/plugins/inspectorViews/styles/StylesInspectorViewProvider.js [hunk 3]
```diff
@@ -46,14 +46,17 @@ export default function StylesInspectorViewProvider(openmct) {
...
+      const domainObject = objectContext?.item;
+      const isFlexibleLayoutContainer =
+        domainObject?.type === 'flexible-layout' && objectContext.type === 'container';
 
       if (layoutItem) {
         return true;
       }
 
-      if (!domainObject) {
+      if (!domainObject || isFlexibleLayoutContainer) {
```
Comment (khalidadil) [inline src/plugins/inspectorViews/styles/StylesInspectorViewProvider.js:59]: Added to disable the styles tab on flexible layout containers

File: src/plugins/plot/stackedPlot/StackedPlot.vue [hunk 4]
```diff
@@ -23,7 +23,7 @@
 <template>
   <div
     v-if="loaded"
-    class="c-plot c-plot--stacked holder holder-plot has-control-bar"
+    class="c-plot c-plot--stacked holder holder-plot has-control-bar u-style-receiver js-style-receiver"
```
Comment (khalidadil) [inline src/plugins/plot/stackedPlot/StackedPlot.vue:26]: Added to fix the styles landing on the plot instead of the container. This created an issue where removing styles wasn't working properly since the add and remove targeted different elements.

File: src/ui/layout/BrowseBar.vue [hunk 5]
```diff
@@ -362,6 +362,10 @@ export default {
         title: 'Saving'
       });
 
+      const currentSelection = this.openmct.selection.selected[0];
+      const parentObject = currentSelection[currentSelection.length - 1];
+      this.openmct.selection.select(parentObject);
```
Comment (khalidadil) [inline src/ui/layout/BrowseBar.vue:367]: Added to reset selection of views when they're saved.

File: e2e/tests/visual-a11y/styling.visual.spec.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+  test('selecting a flexible layout column hides the styles tab', async ({ page }) => {
+    // Edit Flexible Layout
+    await page.locator('[title="Edit"]').click();
+    // Expect to find a styles tab
+    let stylesTab = await page.locator('.c-inspector__tab[title="Styles"]').count();
+    expect(stylesTab).toBe(1);
+
+    // Select flexible layout column
+    await page.locator('.c-fl-container__header').first().click();
```
Comment (unlikelyzero) [inline e2e/tests/visual-a11y/styling.visual.spec.js:102]: these are great tests. I need to update the locator strategy and it may involve some changes to the flexibleLayout component props

File: e2e/tests/visual-a11y/styling.visual.spec.js (hunk 1 above)
Comment (unlikelyzero) [reply e2e/tests/visual-a11y/styling.visual.spec.js:26]: no this is great

File: .git-blame-ignore-revs [hunk 7]
```diff
@@ -4,6 +4,10 @@
 # Requires Git > 2.23
 # See https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt
 
+# vue-eslint update 2019
+14a0f84c1bcd56886d7c9e4e6afa8f7d292734e5
+# eslint changes 2022
+d80b6923541704ab925abf0047cbbc58735c27e2
```
Comment (unlikelyzero) [inline .git-blame-ignore-revs:10]: drive-by to make it easier to work with flex layouts

File: e2e/tests/functional/styling.e2e.spec.js [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+    await page.goto('./', { waitUntil: 'domcontentloaded' });
+
+    // Create a Flexible Layout and attach to the Stacked Plot
+    flexibleLayout = await createDomainObjectWithDefaults(page, {
+      type: 'Flexible Layout',
+      name: 'Flexible Layout'
+    });
+  });
+
+  test('selecting a flexible layout column hides the styles tab', async ({ page }) => {
```
Comment (unlikelyzero) [inline e2e/tests/functional/styling.e2e.spec.js:42]: broke this out to its own suite to improve runtime and improve the opportunity to add new styling specific inspector tests

File: e2e/tests/functional/styling.e2e.spec.js [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+    await page.goto(flexibleLayout.url, { waitUntil: 'domcontentloaded' });
+
+    // Edit Flexible Layout
+    await page.getByLabel('Edit').click();
+
+    // Select styles tab
+    await page.getByRole('tab', { name: 'Styles' }).click();
+
+    // Set styles using setStyles function
+    await setStyles(page, backgroundColor, textColor, page.locator('.c-fl__container-holder'));
```
Comment (unlikelyzero) [inline e2e/tests/functional/styling.e2e.spec.js:111]: we'll need to figure out why the `page.locator('.c-fl__container-holder'));` locator is the only way to select the holder

File: e2e/tests/functional/styling.e2e.spec.js [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+    await page.goto(flexibleLayout.url, { waitUntil: 'domcontentloaded' });
+
+    // Edit Flexible Layout
+    await page.getByLabel('Edit').click();
+
+    // Select styles tab
+    await page.getByRole('tab', { name: 'Styles' }).click();
+
+    // Set styles using setStyles function
+    await setStyles(page, backgroundColor, textColor, page.getByLabel('Stacked Plot Frame'));
```
Comment (unlikelyzero) [inline e2e/tests/functional/styling.e2e.spec.js:152]: This is working here

File: e2e/tests/functional/styling.e2e.spec.js [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+
+    // Check styles using checkStyles function
+    await checkStyles(
+      page,
+      hexToRGB(backgroundColor),
+      hexToRGB(textColor),
+      page.getByLabel('Stacked Plot Frame')
```
Comment (unlikelyzero) [inline e2e/tests/functional/styling.e2e.spec.js:159]: but not here. I'm thinking is something fundamentally wrong in the way we are selecting elements for comparison.

File: package.json [hunk 12]
```diff
@@ -111,7 +111,7 @@
     "test:e2e:visual:ci": "percy exec --config ./e2e/.percy.ci.yml --partial -- npx playwright test --config=e2e/playwright-visual-a11y.config.js --project=chrome --grep-invert @unstable",
     "test:e2e:visual:full": "percy exec --config ./e2e/.percy.nightly.yml -- npx playwright test --config=e2e/playwright-visual-a11y.config.js --grep-invert @unstable",
     "test:e2e:full": "npx playwright test --config=e2e/playwright-ci.config.js --grep-invert @couchdb",
-    "test:e2e:watch": "npx playwright test --ui --config=e2e/playwright-ci.config.js",
+    "test:e2e:watch": "npx playwright test --ui --config=e2e/playwright-watch.config.js",
```
Comment (unlikelyzero) [inline package.json:114]: drive-by: improve the ability to debug multiple tests at once

File: src/plugins/flexibleLayout/components/ContainerComponent.vue [hunk 13]
```diff
@@ -25,13 +25,14 @@
     class="c-fl-container"
     :style="[{ 'flex-basis': sizeString }]"
     :class="{ 'is-empty': !frames.length }"
-    role="group"
-    :aria-label="`Container ${container.id}`"
+    role="grid"
```
Comment (unlikelyzero) [inline src/plugins/flexibleLayout/components/ContainerComponent.vue:28]: this is technically a grid object

File: src/plugins/flexibleLayout/components/ContainerComponent.vue [hunk 14]
```diff
@@ -25,13 +25,14 @@
...
   >
     <div
       v-show="isEditing"
       class="c-fl-container__header"
       draggable="true"
+      role="columnheader"
+      :aria-label="`Container Handle ${index}`"
```
Comment (unlikelyzero) [inline src/plugins/flexibleLayout/components/ContainerComponent.vue:35]: I believe that this is the correct handle

File: src/plugins/flexibleLayout/components/ContainerComponent.vue [hunk 15]
```diff
@@ -44,12 +45,14 @@
...
     />
 
-    <div class="c-fl-container__frames-holder">
+    <div role="row" class="c-fl-container__frames-holder">
       <template v-for="(frame, i) in frames" :key="frame.id">
         <frame-component
           class="c-fl-container__frame"
           :frame="frame"
           :index="i"
+          role="gridcell"
```
Comment (unlikelyzero) [inline src/plugins/flexibleLayout/components/ContainerComponent.vue:54]: this becomes a cell in a grid

File: src/plugins/flexibleLayout/components/ContainerComponent.vue [hunk 16]
```diff
@@ -44,12 +45,14 @@
...
+          :aria-label="`Container Frame ${index}`"
```
Comment (unlikelyzero) [inline src/plugins/flexibleLayout/components/ContainerComponent.vue:55]: accessible by the frame index

File: src/ui/toolbar/ToolbarContainer.vue [hunk 17]
```diff
@@ -20,7 +20,7 @@
  at runtime from the About dialog for additional information.
 -->
 <template>
-  <div class="c-toolbar">
+  <div role="menubar" class="c-toolbar">
```
Comment (unlikelyzero) [inline src/ui/toolbar/ToolbarContainer.vue:23]: added for the ability to select an object in the flexible layout

File: src/ui/toolbar/components/ToolbarButton.vue [hunk 18]
```diff
@@ -24,6 +24,8 @@
     <div
       ref="button"
       class="c-icon-button"
+      role="menuitem"
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarButton.vue:27]: can select as a menu item by the title of the option

File: src/ui/toolbar/components/ToolbarColorPicker.vue [hunk 19]
```diff
@@ -21,23 +21,26 @@
...
       @click="handleClick"
     >
       <div
         class="c-swatch"
-        :style="{
-          background: options.value
-        }"
+        :style="{ background: options.value }"
+        role="img"
+        :aria-label="None"
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarColorPicker.vue:35]: this is an image which needs to be neutered with a none selector

File: src/ui/toolbar/components/ToolbarColorPicker.vue [hunk 20]
```diff
@@ -21,23 +21,26 @@
 -->
 <template>
   <div class="c-ctrl-wrapper">
-    <div
+    <button
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarColorPicker.vue:24]: this has always been a button

File: src/ui/toolbar/components/ToolbarColorPicker.vue [hunk 21]
```diff
@@ -21,23 +21,26 @@
...
+        :aria-label="None"
       ></div>
-    </div>
+    </button>
     <div v-if="open" class="c-menu c-palette c-palette--color">
       <div
         v-if="!options.preventNone"
         class="c-palette__item-none"
+        role="grid"
+        aria-label="No Style"
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarColorPicker.vue:43]: Selectable with 'No Style'

File: src/ui/toolbar/components/ToolbarColorPicker.vue [hunk 22]
```diff
@@ -47,8 +50,11 @@
         <div
           v-for="(color, index) in colorPalette"
           :key="index"
+          role="gridcell"
           class="c-palette__item"
           :style="{ background: color.value }"
+          :title="color.value"
+          :aria-label="color.value"
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarColorPicker.vue:57]: presents the hex value of the color to the user. Given that this is a very "visual" component, I assume that the screenreader would know what to do with this

Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarColorPicker.vue:57]: presents the hex value of the color to the user. Given that this is a very "visual" component, I assume that the screenreader would know what to do with this

File: src/ui/toolbar/components/ToolbarSeparator.vue [hunk 23]
```diff
@@ -20,7 +20,7 @@
  at runtime from the About dialog for additional information.
 -->
 <template>
-  <div class="c-toolbar__separator"></div>
+  <div role="separator" class="c-toolbar__separator"></div>
```
Comment (unlikelyzero) [inline src/ui/toolbar/components/ToolbarSeparator.vue:23]: this is a separator. We should add some stubs for separator tests

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Almost there (in terms of tests). Something is fundamentally wrong in the way that container objects are selected by the user. It should be as simple as:

`page.getByLabel('Stacked Plot Frame')`

But it's not. We should figure out how object selection is being handled and make sure that we know what the user is intending to change when the select objects in the flex layout

**Comment by khalidadil** [comment]
> Almost there (in terms of tests). Something is fundamentally wrong in the way that container objects are selected by the user. It should be as simple as:
> 
> `page.getByLabel('Stacked Plot Frame')`
> 
> But it's not. We should figure out how object selection is being handled and make sure that we know what the user is intending to change when the select objects in the flex layout

@ozyx Made some great fixes for this. The issue is that the styling is happening at the plugin component level, but clicks are being captured at the frame level. So we're now clicking at the frame level, setting the style, and then comparing the style at the component level. This allows us to not use `force: true` and verify the styles are correct in the plugin.

**Comment by unlikelyzero** [comment]
This is great. 

I think the plan here is to scope the changes to what @andrew mentioned in the 9/18 meeting. At that time, we can probably fix the sub object selection and styling logic while keeping the test coverage.

In reviewing the issue, it looks like we don't have total coverage for the scenarios that Charles outlined. Let me take another swing at the tests

File: .git-blame-ignore-revs (hunk 7 above)
Comment (davetsay) [reply .git-blame-ignore-revs:10]: this is so cool

File: src/ui/layout/BrowseBar.vue (hunk 5 above)
Comment (davetsay) [reply src/ui/layout/BrowseBar.vue:368]: Isn't the desired behavior to stay on the saved object?

File: src/api/menu/components/SuperMenu.vue [hunk 24]
```diff
@@ -88,6 +93,12 @@ export default {
       hoveredItem: {}
     };
   },
+  computed: {
+    getLabel() {
```
Comment (davetsay) [inline src/api/menu/components/SuperMenu.vue:97]: this is super nitpicky but `get` to me implies a method call

File: src/api/menu/components/MenuComponent.vue [hunk 25]
```diff
@@ -65,6 +65,12 @@
 import popupMenuMixin from '../mixins/popupMenuMixin';
 export default {
   mixins: [popupMenuMixin],
-  inject: ['options']
+  inject: ['options'],
+  computed: {
+    getLabel() {
```
Comment (davetsay) [inline src/api/menu/components/MenuComponent.vue:70]: same nitpick (working backwards). `get` to me implies a method call.

**Review by davetsay — CHANGES_REQUESTED** [review]
some super tiny changes. great work.

File: src/api/menu/components/SuperMenu.vue (hunk 24 above)
Comment (khalidadil) [reply src/api/menu/components/SuperMenu.vue]: Makes sense! I changed this to `optionsLabel`.

File: src/api/menu/components/MenuComponent.vue (hunk 25 above)
Comment (khalidadil) [reply src/api/menu/components/MenuComponent.vue]: This one too

File: src/ui/layout/BrowseBar.vue (hunk 5 above)
Comment (davetsay) [reply src/ui/layout/BrowseBar.vue:368]: ok. thanks for explaining this is for when you edit objects in display layouts, you should select the layout after saving. that makes sense.

If you aren't in a display layout, the code will select itself as the parent, and selection api takes care of selecting the same selection so things aren't getting re-mounted.

**Review by davetsay — APPROVED** [review]
LGTM

**Comment by unlikelyzero** [comment]
From the issue:
- [x] The Styles tab MUST recognize if an element can be styled or not. If it can't, it must have a disabled or other state that includes a message like "This element cannot be styled".
- [x] The overall Flex Layout or Stacked Plot itself MUST be style-able.
- [x] Individual child elements of Flex Layouts and Stacked Plots (telem points, child Overlay Plots, etc.) MUST BE STYLE-ABLE. 
- [ ] If an overall style has been applied to the parent layout or plot, the individual styling should be able to coexist with that.
- [x] Flex Layout containers should NOT be style-able.
- [x] When the user chooses the "no style" option, REMOVE THE STYLE TAG - do not add a style attribute with a hard-coded color.
- [ ] Conditional Styling MUST work everywhere that static styling does in the same way it does elsewhere.
- [ ] Users MUST be able to click on the empty area above the plot and legend areas to select the parent Stacked Plot container.

File: e2e/helper/stylingUtils.js [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+const { expect } = require('../pluginFixtures');
```
Comment (ozyx) [inline e2e/helper/stylingUtils.js:23]: We can do this now:
```suggestion
import { expect } from '../pluginFixtures.js';
```

File: e2e/helper/stylingUtils.js [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Converts a hex color value to its RGB equivalent.
+ *
+ * @param {string} hex - The hex color value.
```
Comment (ozyx) [inline e2e/helper/stylingUtils.js:28]: We should mention the expected format briefly

File: e2e/helper/stylingUtils.js [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+ */
+async function setStyles(page, borderColorHex, backgroundColorHex, textColorHex, locator) {
+  await locator.click(); // Assuming the locator is clickable and opens the style setting UI
+  await page.getByLabel('Set border color').click();
+  await page.getByLabel(borderColorHex).click();
+  await page.getByLabel('Set background color').click();
+  await page.getByLabel(backgroundColorHex).click();
+  await page.getByLabel('Set text color').click();
+  await page.getByLabel(textColorHex).click();
+}
```
Comment (ozyx) [inline e2e/helper/stylingUtils.js:55]: this rocks

File: e2e/helper/stylingUtils.js [hunk 29]
```diff
@@ -0,0 +1,88 @@
...
+  expect(layoutStyles.background).toContain(expectedBackgroundColor);
+  expect(layoutStyles.fontColor).toContain(expectedTextColor);
+}
+
+// eslint-disable-next-line no-undef
+module.exports = {
+  checkStyles,
+  setStyles,
+  hexToRGB
+};
```
Comment (ozyx) [inline e2e/helper/stylingUtils.js:88]: We can do this now:
```suggestion
export {
  checkStyles,
  setStyles,
  hexToRGB
};
```

File: e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js [hunk 30]
```diff
@@ -0,0 +1,254 @@
...
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/**
+ * This test is dedicated to test styling of stacked plots
+ */
+
+const { test, expect } = require('../../../../pluginFixtures');
+const { createDomainObjectWithDefaults } = require('../../../../appActions');
+const { hexToRGB, setStyles, checkStyles } = require('../../../../helper/stylingUtils');
```
Comment (ozyx) [inline e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js:29]: Update to ESM syntax

File: e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js [hunk 31]
```diff
@@ -0,0 +1,254 @@
...
+
+const setBorderColor = '#ff00ff';
+const setBackgroundColor = '#5b0f00';
+const setTextColor = '#e6b8af';
+const defaultFrameBorderColor = '#e6b8af'; //default border color
+const defaultBorderTargetColor = '#aaaaaa';
+const defaultTextColor = '#aaaaaa'; // default text color
+const inheritedColor = '#aaaaaa'; // inherited from the body style
+const pukeGreen = '#6aa84f'; //Ugliest green known to man
```
Comment (ozyx) [inline e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js:38]: Love it

**Review by ozyx — CHANGES_REQUESTED** [review]
Looks great! We should merge in the changes from `master` and update our imports/exports to be ESM syntax. Other than that, LGTM!

File: e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js [hunk 32]
```diff
@@ -0,0 +1,211 @@
...
+    await page.getByRole('listitem', { name: 'Save and Finish Editing' }).click();
+
+    await checkStyles(
+      hexToRGB(setBorderColor),
+      hexToRGB(setBackgroundColor),
+      hexToRGB(setTextColor),
+      page.getByLabel('Stacked Plot Style Target')
+    );
+
+    await checkFontStyles(setFontSize, setFontType, page.getByLabel('Stacked Plot Item Sine Wave Generator 1'));
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js:110]: checkFontStyles isn't able to find anything for 'font-family' on the two objects that we're checking for

File: e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js [hunk 33]
```diff
@@ -0,0 +1,211 @@
...
+
+    // Reload page
+    await page.reload({ waitUntil: 'domcontentloaded' });
+
+    //Verify styles are correct after reload
+    await checkStyles(
+      hexToRGB(setBorderColor),
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/styling/stackedPlotStyling.e2e.spec.js:117]: This is returning another color entirely which makes me think we have some sort of bug. We don't need to fix it in this PR (and disable a portion of this test), but I'd like @khalidadil to take a look

**Review by unlikelyzero — APPROVED** [review]
